File: MaximumSubArray.java
//https://leetcode.com/problems/maximum-subarray/

class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int currentSum = 0;

        for (int i = 1; i < nums.length; i++) {
            currentSum = Math.max(nums[i], currentSum + nums[i]);
            maxSum = Math.max(maxSum, currentSum);
        }

        return maxSum;

    }
}

File: Medium/MaximumSubArray/MaximumSubArray.py
# https://leetcode.com/problems/maximum-subarray/

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0] 
        curSum = 0 
        for num in nums:
            if curSum < 0:
                curSum = 0
            curSum = curSum + num
            maxSum = max(maxSum , curSum)
        return maxSum